ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/DdlDmlResponseHandlers.java

```diff
@@ -68,6 +72,13 @@ private static boolean isCommandStatusEntity(final JsonObject ksqlEntity) {
         && ksqlEntity.getJsonObject("commandStatus") != null;
   }
 
+  private static boolean isIfNotExistsWarning(final JsonObject ksqlEntity) {
+    return ksqlEntity.getString("statementText") != null
+        && ksqlEntity.getString("statementText").contains("IF NOT EXISTS")
+        && ksqlEntity.getString("message") != null
+        && ksqlEntity.getString("message").contains("A stream with the same name already exists.");
```

Comment:
This could also apply to tables and connectors. If there's some way to determine if the ksqlEntity is a warning rather than an error, then that would be the most inclusive way to catch everything.

Reply:
good call

Reply:
There's also `DROP STREAM|TABLE|CONNECTOR IF EXISTS` which also returns a warning. I think just checking for the warning and making sure it's not an error should be enough.

Reply:
Are you saying don't check the msg contents?

Reply:
Yeah. As long as we check after trying every other entity type, it should be accurate. It also definitely won't overlap with `ErrorEntity` because those have a field called `errorMessage` instead of just `message`.

Reply:
sadly we don't check the Entity types and we just check to see if it has a cmd seq. If it doesn't we consider it an error. Since the entire point is avoid putting these on the cmd topic is doesn't really work

Reply:
ohh okay. this solution works to just check for the `IF NOT EXISTS` case,  but`WarningEntity` is also used for `DROP CONNECTOR IF EXISTS` and `UNDEFINE` (neither of those should be accepted by the `executeStatement` method, but they get executed anyway.)

Reply:
Maybe we can do that in a follow up. I think we should keep this targeted to avoid secondary effects for this release